pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1138,6 +1319,21 @@ func handleCreate(ctx *domainContext, key string, config *types.DomainConfig) {
 		VmConfig:           config.VmConfig,
 		Service:            config.Service,
 	}
+
+	ctx.resourcesUsedByVMs[config.UUIDandVersion.UUID] = ResourcesUsedByVM{CPUSet: make(map[int]bool)}
+	// Not to get races with the other threads, that try to assign the CPUs at the moment, we have to lock here
+	// Will release after the corresponding control files are created and the arguments are passed to containerd
+	ctx.vmResourcesLock.Lock()
+	defer ctx.vmResourcesLock.Unlock()
+	err := assignCPUs(ctx, config)
```

Comment:
Don't we also need some unassignCPUs when the domainconfig is deleted?
That seems to be missing. 

For other resources like IO adapters we do the assign and free in the Activate and Inactivate paths. This enables temporarl reuse. The user can create multiple app instances with pinned CPU and then boot (aka activate) a subset of them.
Then when they are halted (aka inactivate completes) the pinned CPUs are made available to other app instances.

If you do it that way it also means that if handleModify gets a different number of vCPUs, it can apply the change by pinning a different set of CPUs (smaller or larger). That presumably requires a restart the same way the number of vpus require a restart as checked in https://github.com/lf-edge/eve/blob/d57dc774a6f2f6c776e95ec8f065709964629c3b/pkg/pillar/cmd/zedmanager/zedmanager.go#L869

Reply:
unassignCPUs is done in the scope of this commit:
https://github.com/lf-edge/eve/pull/2888/commits/34231fbee702a15512a062a0b2b5422afa7901f0

Reply:
Sorry I must have missed that code; maybe I was assuming a particular naming scheme for the functions. My bad.

But it still makes sense to do this assignment and free in active and inactivate, respectively.

Reply:
Done